### ISSUE TITLE
Clothing/armor modifications should have data entries and be moddable.

### ISSUE THREAD

**Issue by Asmageddon** [CleverRaven/Cataclysm-DDA#19936]
Because there's a lot of great stuff to introduce: Padding with chitin, embedding a thermal circuit, makeshift waterproofing, a nomex layer, extra pockets, extra thickness for sturdiness, etc.

It would all go a long way towards making survivor gear something more obsolete, and more of an implicit result of tinkering than an explicit set of higher-end recipes.

**Comment by Asmageddon** [comment]
Power armor sounds like a whole other mess entirely. My only "wishlist" stuff wrt/ power armor would be to have something like powered and non-powered non-armor industrial exoskeletons that aid with hauling.

**Comment by kevingranade** [comment]
Not a fan of extensively moddable clothing, clothing is already rather OP, and something like this just makes it worse.  We had some more clothing mods available via the tailoring kit, but we pulled them back out because there was no sensical way to determine based on existing item characteristics which mods should be legal or what their effects should be.
The alternative is to add more parameters to clothing definitions to explicitly state what the effects of said modifications would be, but all of a sudden that's a HUGE amount of content to add for an extremely niche feature.
The reason it is niche is simple, ad-hoc additions of features like padding, pockets, insulation, etc, should never be as good as purpose-built clothing with the same features, so the balance point to aim for with clothing modifications is to basically make them irrelevant since it's not that hard to find the superior clothing that just has the properties you want

**Comment by Asmageddon** [comment]
Survivor clothing is already far superior to any existing gear, and the MBR vest is very lightweight and powerful, for being craftable without a recipe of any kind.

I think that if cooking or electronics 10 means being on the level of crafting mutagenic serums and advanced bionics, it would be reasonable for higher level tailoring to allow one to create clothing comparable or in specific cases superior to pre-existing articles.

I think that a lot of the balance could be accomplished with filtering what clothing you can put a given mod on by materials, slots, and layers, so you cannot reinforce underwear with kevlar, etc.

I really think that it's a far superior alternative to having a set of stupidly OP survivor gear.

**Comment by kevingranade** [comment]
On Jan 12, 2017 2:07 AM, "Asmageddon" <notifications@github.com> wrote:

Survivor clothing is already far superior to any existing gear, and the MBR
vest is very lightweight and powerful, for being craftable without a recipe
of any kind.

Tiny nit, the MBR vest shouldn't be craftable with no recipe, that's a
really complicated article of clothing.  It's so good because it happens to
designed to deliver what a survivor needs.

I think that if cooking or electronics 10 means being on the level of
crafting mutagenic serums and advanced bionics, it would be reasonable for
higher level tailoring to allow one to create clothing comparable or in
specific cases superior to pre-existing articles.

That doesn't contradict what I said, the question is which is going to be
better, existing clothes modded with various extensions or clothing
tailored to incorporate those elements.  IMO it's clear that clothing
designed to fill a niche is going to be superior to clothing patched to
make it fit that niche.  This is independent of how good survivor gear is
relative to found clothes, modded clothes will always be worse.  Therefore
the niche for modded clothes is tiny and doesn't justify the apparently
large effort to make them fit in that niche.

I think that a lot of the balance could be accomplished with filtering what
clothing you can put a given mod on by materials, slots, and layers, so you
cannot reinforce underwear with kevlar, etc.

Last time it was an abject failure, but you can certainly give it a try.

I really think that it's a far superior alternative to having a set of
stupidly OP survivor gear.

It's not going to to be an alternative, my position is that for any
combination of clothes and mods, there should in general be a custom made
article that is better.

**Comment by Coolthulhu** [comment]
> It's not going to to be an alternative, my position is that for any combination of clothes and mods, there should in general be a custom made article that is better.

That goes against the whole scavenge vs craft, though.
Plus, it clashes with limitations on crafting of other complex objects, such as guns.
I'm pretty sure improving on design of modern combat armor is harder than crafting an AK-74 knockoff in a shed.

**Comment by kevingranade** [comment]
I think there's still some msunderstanding here, let me break it down,
there are three categories of clothing we're talking about here:
Pre-cataclysm clothing - well-made and designed in general, but only rarely
(MBR vest, SWAT gear) optimized for the kind of scenario the survivor finds
themselves in.
Pre-cataclysm clothing with modifications - Extra insulation, padding,
leather covering, etc, this adds some desireable properties to the
clothing, but does so ineficiently, adding excess bulk or stiffness.
Post-cataclysm purpose-built clothing -  This is clothing tailored from the
ground up to meet the needs of a survivor in the cataclysm, it meets the
survivor's needs and doesn't have the drawbacks of having certain features
retrofitted onto it.

My assertion is that the second category is squezed very tightly in between
the other two categories.  The current system allows some simple
modifications, but an extensive patching system isn't worthwhile because
it's extremely difficult to balance (combinatoric explosion of options) and
will be quickly overshadowed by custom-made clothing items.

> That goes against the whole scavenge vs craft, though.

I'm not clear on what you mean, scavenge versus craft is a tradeoff, not a
rule.  The categories I outline above are roughly a continuum between
scavenge and crafting.  How good various articles are is only weakly
related to where on the continum they are, but the best of the third
category should always be better than the best of the second category, and
sometimes better than the best of the first category.  (power armor is an
obvious counter-example, you just can't make it)

> I'm pretty sure improving on design of modern combat armor is harder than
crafting an AK-74 knockoff in a shed.

There're three problems with this statement, one, you shouldn't be able to
craft a knockoff AK-74 in a shed, so it's not a valid example.
Two, I'd argue that modern combat armor focuses on ballistic protection
over impacts and cuts/stabs, and armor that does focus on impact protection
is designed for sports where impacts are incidental, not targeted,
basically nothing focuses on cutting or stabbing protection.  Modern body
armor designed to protect solely against an advesary that attacks by
bludgeoning, striking, biting, etc either doesn't exist or is
extrordinarally rare, so it does seem feasible to make armor that fills
that niche better than anything available.
Three, the discussion at hand is about patching clothing versus crafting
new clothing, the "crafting cap" doesn't enter into it.

**Comment by Asmageddon** [comment]
The way I see it, we have lining and padding. Add a third category for stuff like nomex/plastic covering or pockets, allow only one mod per article per class, and focus on utility mods(warmth, pockets) that either improve clothes just a bit, or increase encumbrance severely(kevlar/steel/whatever padding).

Purpose-built clothing should be superior to other modded articles, yes, but that doesn't mean it shouldn't be possible to improve it, even if just a bit, or to improve other, less "high end" pieces.

You shouldn't be able to mod a shirt into body armor, but you should be able to add pockets, a warmer circuit, maybe some leather padding to some random suit.

Also, the way I see it, having data entries means there can be a mod for this stuff, so you don't have to have superalloy padding and spiked cover in vanilla, but whoever wants to, can.

**Comment by kevingranade** [comment]
> Add a third category for stuff like nomex/plastic covering or pockets, allow only one mod per article per class, and focus on utility mods(warmth, pockets) that either improve clothes just a bit, or increase encumbrance severely(kevlar/steel/whatever padding).

That's just the thing, lining (warmth) and padding (protection) is as far as I want it to go.  If you want to extract that to a data-based definition and then add to it in a mod, great go ahead, but if you want to extend how much this code does with this in vanilla, you're going to have an uphill battle.  Adding pockets in particular is usually nonsensical and impossible to balance properly because carrying capacity is so valuable and there are so many articles of clothing you could add it to.

> Purpose-built clothing should be superior to other modded articles, yes, but that doesn't mean it shouldn't be possible to improve it, even if just a bit, 

Actually it should not be possible to improve it further, effectively what you're saying is that any time someone designs an article of clothing, there is always room for improvement.  If it's purpose-built to be as good as it can be, no you can't just slap more patches on it and improve it.

**Comment by Asmageddon** [comment]
What I'm saying is that most clothing was built for a single purpose, and often balanced between quality and cost of production. Look at how devoid of pockets women's clothing tends to be, or how encumbering denim jeans are compared to tracksuit jeans. You obviously should not be able to improve a piece of body armor(without paying a hefty price, at least), but IMO ordinary clothing is fair game.

Also, carrying capacity on most articles has 1:1 cost with encumbrance, exceptions being a few military articles(drop pouches, that one other thing I forgot), and survivor clothing. IMO if adding pockets had 1.5x the cost, it wouldn't be (that) bad, it'd just allow you to squeeze some extra storage out of clothes on body parts on which you can accept some more encumbrance.
